app/models/user.rb

```diff
@@ -13,6 +15,7 @@ class User < ActiveRecord::Base
     after_initialize do
         self.secret ||= SecureRandom.base64(24)
         self.username ||= UsernameGenerator.new
+        self.avatar ||= Identicon.data_url_for self.secret, 256, [255, 255, 255]
```

Comment:
Is there any reason this is based on `self.secret` rather than username/email/username.concat(email)?  

(the security purist in me is acting up a bit, sorry)